contracts/p0/StRSR.sol

```diff
@@ -524,6 +524,7 @@ contract StRSRP0 is IStRSR, ComponentP0, EIP712Upgradeable {
     }
 
     function setRewardRatio(uint192 val) public governance {
+        _payoutRewards();
```

Comment:
Here we want to do a similar thing and only payout rewards if not frozen

Reply:
Possibly cleanest is making `payoutRewards()` public and calling it from here.

Reply:
I can handle this, though, once the tests pass

Reply:
the internal `_payoutRewards()` does not care if the system is paused or frozen, only the external call has the modifier

Reply:
Right but we should not hand out any rewards while frozen, even if governance changes the ratio

Reply:
Sorry, my previous suggestion was bad. We should just do `if (!main.frozen()) _payoutRewards()` and nothing more, I think